crates/base-db/src/semantics/tex.rs

```diff
@@ -55,6 +56,20 @@ impl Semantics {
                     latex::COMMENT if token.text().contains("texlab: ignore") => {
                         self.diagnostic_suppressions.push(token.text_range());
                     }
+                    latex::COMMENT if token.text().contains("warnings: off") => {
```

Comment:
I think that `texlab: warnings off` and `texlab: warnings on` would be more consistent with the existing magic comment (`texlab: ignore`).